feat(project-funds): add initial version of project funds (#999)
openjs-foundation/cross-project-council: project-resources/PROJECT_FUNDS.md

# OpenJS Foundation Project-specific Initiative Funds

> The OpenJS Foundation provides a set of monetary funds for projects of the OpenJS Foundation. 

Reviewer: ```suggestion
The OpenJS Foundation (OpenJSF) provides a set of monetary funds for projects of the OpenJSF. 
```

These funds are designed to support collaboration and incentive project-specific initiatives that have high-impact within the individual project's scope.

Alongside these lines, the fund exist to support the development of these 

> initiatives and give visibility/recognition to the champions of these project-specific initiatives, highligthing the individuals behind them.

Reviewer: > and give visibility/recognition to the champions of these project-specific initiatives, highligthing the individuals behind them.

I don't think that should be the purpose of this fund. I would suggest simplifying that section (or anchoring back to principles that aren't subject to a controversy.).

Author: Hmm, that's what I've understood from what @mcollina wrote on the original issue. Maybe bad phrasing from my side?

Reviewer 2: I don't think I ever said that.
In fact, it's the other way - funds will be needed where our volunteer-based model fails.

Author: @mcollina it is from this paragraph that I took what I wrote " Volunteers are driven by their motives but usually require visibility/recognition of their work. However, this model fails for tasks without recognition: if you do your job well, no one would know you did it."

I might have misunderstood or just poorly phrased my statements. Could you suggest an alternative patch here for the PR? Thank you!

Reviewer: Regardless, I don't think there's a benefit in attaching a value statement of that nature to the disbursement of funds.

Author: I'm fine with any path, just tried to give an introductory text about what the project funds are and why they exist. Sounds like it was not a good text, if anybody has a better suggestion please comment, I'd love to have this aligned with what we believe that fits better here 🙃

Reviewer: Didn't mean to be negative about your input. Apologies if it came across that way. I'm very thankful that you got this started.

Author: Oh no, it's fine, I just honestly don't know what we should replace the current entry with 👀

Author: I believe Matteo already applied a patch here 🎉

> Alongside these lines, the fund exist to support the development of these 
> initiatives and give visibility/recognition to the champions of these project-specific initiatives, highligthing the individuals behind them.

Reviewer 2: ```suggestion
Alongside these lines, the fund exists to support the development of the project when the volunteer-based approach is not enough to support the community, e.g. infrastructure, tools, or security.
```

Author: Thank you for the patch, @mcollina!

## F.A.Q.

#### How might a project request for funds?

> Funds should be requested on an ad-hoc basis through a PR requesting the funds. The requests do not need to follow a specific template,

Reviewer: FWIW, I think offering a template makes everyone's life easier, so I think we just should do that.

Author: Makes sense. WDYT the template should contain?

Author: Which points, the ones @joesepi wrote?

Reviewer: My understanding is @erickzhao was referring those in the quoted sentence:

> To access the funds, the project would need to prepare a 1-pager plan, including the purpose and type fo activity, success criteria, timeline, and funds requested

Author: Oh my goodness, how I missed that! Sorry @erickzhao 🙈

Reviewer: There seemed to be agreement to have a template for this. Can we bake it into this PR?

Author: Sure thing!

but the description should contain relevant information regarding the project initiative, 
and any information that might be useful to highlight the need of funds and their benefits.

#### How these requests get evaluated?

> The requests should go through the OpenJSF Cross Project Council (CPC) and must have approval from the CPC members. 

Author: ```suggestion
The request must be made to the CPC for evaluation. The CPC evaluates the request following its [decision-making process](https://github.com/openjs-foundation/cross-project-council/blob/main/CPC-CHARTER.md#section-9-decision-making) and makes a recommendation to the Board.
```

> The requests should go through the OpenJS Foundation CPC (Cross Project Council) and must have approval from the CPC members. 

Author: It would be interesting to know if CPC voting must be unilateral, or how exactly the voting works.

I would also suggest for us to have a Markdown file (Basically a table) like we have for Travel Funds to keep documented previous "approved project funds requests".

Reviewer 3: Do we have existing approval/voting processes we can fall back on?

Author: The only similar process I'm aware of is the travel fund...

Reviewer: > It would be interesting to know if CPC voting must be unilateral, or how exactly the voting works.

_"The request must be made to the CPC for evaluation. The CPC evaluates the request following its [decision-making process](https://github.com/openjs-foundation/cross-project-council/blob/a49a8d2e04716525116c0ef787e1187d4dcd9e3e/CPC-CHARTER.md#section-9-decision-making) and makes a recommendation to the Board."_

> The requests should go through the OpenJS Foundation CPC (Cross Project Council) and must have approval from the CPC members. 

Reviewer: ```suggestion
The requests should go through the OpenJSF Cross Project Council (CPC) and must have approval from the CPC members. 
```

Finally the request should be forwarded to the Board of Directors for final approval.

#### How much might a project request?

> Dollar amounts will be determined on ad-hoc basis (proposed cap isn't necessary). Requests should be reasonable and can be discussed.

Author: It would be interesting to have a range of what is feasible to be asked. And if different project categories (At-large, Impact, Incubation) should have different limits.

Reviewer 3: > And if different project categories (At-large, Impact, Incubation) should have different limits.

Would a note clarifying that project status will affect the allocated budget help?

Reviewer: > Would a note clarifying that project status will affect the allocated budget help?

I think that's not necessarily true and also not super useful.

Reviewer: I don't think we have a good answer for this right now. :(

> Dollar amounts will be determined on ad-hoc basis (proposed cap isn't necessary). Requests should be reasonable and can be discussed.

Reviewer: ```suggestion
The level of funding level will be determined on case per case basis after consideration of the request and discussion with the project maintainers.
```

# OpenJS Foundation Project-specific Initiative Funds

The OpenJS Foundation (OpenJSF) provides a set of monetary funds for projects of the OpenJSF. 

> These funds are designed to support collaboration and incentive project-specific initiatives that have high-impact within the individual project's scope.

Author: > Templates work well for issues, but they're clumsy for PRs. Even more so, a PR needs to effect some change.

I wonder if requests should be opened as an issue then? The idea behind a PR is so that either:

- A very visible way of voting against something
- The file change would maybe be a document we use for "logging/auditing" past approved funds, similar to how we do with expense requests for travel funds.

I'm open for suggestions of course :)

> These funds are designed to support collaboration and incentive project-specific initiatives that have high-impact within the individual project's scope.

Reviewer 4: PR is better than issue, because it allows for tracking approval much better and leaves an explicit trace in the repo when merged.

As I mention in my previous comment, my suggestion would be to imitate our pre-existing process on travel funds.

> These funds are designed to support collaboration and incentive project-specific initiatives that have high-impact within the individual project's scope.

Author: Hmm, so we are on the same page, you'd like me to add to this PR a Markdown file with a table similar to the one from the Travel Expenses one?

Just trying to understand what are the next steps here :)

> These funds are designed to support collaboration and incentive project-specific initiatives that have high-impact within the individual project's scope.

Reviewer 4: Yes. We already have one process for asking for funds, and it works. Let's make sure that this process matches that one, rather than creating a different process.

The instructions for requesting travel funds have stood the test of time rather well, let's effectively reuse them here.

> These funds are designed to support collaboration and incentive project-specific initiatives that have high-impact within the individual project's scope.

Author: Hey, @eemeli, I've just pushed the changes! I apologise for the delay.

Alongside these lines, the fund exists to support the development of the project when the volunteer-based approach is not enough to support the community, e.g. infrastructure, tools, or security.

## F.A.Q.

> ### How might a project request for funds?

Reviewer 5: ```suggestion
### How might a project request funds?
```

- **Intended usage of the resouces:**
- **Success Criteria:**
- **Total amount of funds requested:**

## Pre-Request Evaluation

#### Distribution of the Funds

<!-- Explain here how these funds will be distributed/used -->

#### Timeline of the Request

<!-- Explain (possibly with a bullet-list) the timeline of the request, and how/when each part of the funds are being used -->

## Post-request Mortem

<!-- This should be written ad-hoc after the timeline was concluded -->

#### How was the money spent

<!-- How did the money get spent; did it follow the timeline? What diverged? -->

---

- [ ] **Was the success criteria met?**
  - If not, please explain why and if possible any learnings.

cc @openjs-foundation/cpc 
```

### How these requests get evaluated?

The request must be made to the CPC for evaluation. The CPC evaluates the request following its [decision-making process](https://github.com/openjs-foundation/cross-project-council/blob/main/CPC-CHARTER.md#section-9-decision-making) and makes a recommendation to the Board.
Finally the request should be forwarded to the Board of Directors for final approval.

### How much might a project request?

> The level of funding level will be determined on case per case basis after consideration of the request and discussion with the project maintainers.

Reviewer 5: ```suggestion
The level of funding level will be determined on case by case basis after consideration of the request and discussion with the project maintainers.
```
